subprojects/core/src/main/java/org/gradle/execution/plan/OrdinalNodeAccess.java

```diff
@@ -16,32 +16,35 @@
 
 package org.gradle.execution.plan;
 
-import com.google.common.collect.Maps;
-import com.google.common.collect.Streams;
-
 import javax.annotation.Nullable;
-import java.util.List;
+import java.util.IdentityHashMap;
 import java.util.Map;
-import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Streams.concat;
 
 /**
  * A factory for creating and accessing ordinal nodes
  */
 public class OrdinalNodeAccess {
-    private final Map<Integer, OrdinalGroup> groups = Maps.newHashMap();
-    private final Map<OrdinalGroup, OrdinalNode> destroyerLocationNodes = Maps.newHashMap();
-    private final Map<OrdinalGroup, OrdinalNode> producerLocationNodes = Maps.newHashMap();
+    private final OrdinalGroupFactory ordinalGroups;
+    private final Map<OrdinalGroup, OrdinalNode> destroyerLocationNodes = new IdentityHashMap<>();
```

Comment:
```suggestion
    private final IdentityHashMap<OrdinalGroup, OrdinalNode> destroyerLocationNodes = new IdentityHashMap<>();
```

I wonder if the type should be captured in the field as well, since it changes the behavior of the map.

Reply:
Good point.

Reply:
Done.